### ISSUE TITLE
installation fails: node-sass@3.3.3 postinstall: `node scripts/build.js`

### ISSUE THREAD

**Issue by hohwille** [sass/node-sass#1446]
I can not install node-sass. It always fails with the same error (see below).
I read and followed your troubleshooting guide (https://github.com/sass/node-sass/blob/master/TROUBLESHOOTING.md). I updated node and npm to latest versions.
I read through #1170, #1185 and other related issues.
Actually I just wanted to install reveal.js (https://github.com/hakimel/reveal.js#full-setup)
But according to your troubleshooting I also tried:
```
npm cache clean
npm install -ddd node-sass > npm.log 2> npm.err
```

Still I get the same error. To it seems there actually is something wrong with your ```build.js```.
If you need further informations I can provide entire logs or run additional commands.
As you can see I am running on Windows (7 Pro). Maybe it works fine for you as you are using a different OS.

```
npm ERR! Windows_NT 6.1.7601
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\hohwille\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js" "install" "-ddd" "node-sass"
npm ERR! node v5.10.1
npm ERR! npm  v3.8.6
npm ERR! code ELIFECYCLE

npm ERR! node-sass@3.3.3 postinstall: `node scripts/build.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the node-sass@3.3.3 postinstall script 'node scripts/build.js'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the node-sass package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node scripts/build.js
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs node-sass
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls node-sass
npm ERR! There is likely additional logging output above.
npm verb exit [ 1, true ]
```

**Comment by FisherYu-CN** [comment]
Hi, I have the similar problem when I tried to install node-sass on windows xp. I installed VS 2010 and node-gyp should be OK as I can install other node packages. 

```
21511 silly postinstall node-sass@3.4.2 C:\temp\node_modules\.staging\node-sass-2e8f2ecf
21512 info lifecycle node-sass@3.4.2~postinstall: node-sass@3.4.2
21513 verbose lifecycle node-sass@3.4.2~postinstall: unsafe-perm in lifecycle true
21514 verbose lifecycle node-sass@3.4.2~postinstall: PATH: C:\Drossy\Program\nodejs\node_modules\npm\bin\node-gyp-bin;C:\temp\node_modules\node-sass\node_modules\.bin;C:\temp\node_modules\.bin;C:\Drossy\Program\nodejs;C:\Documents and Settings\Administrator\bin;C:\Drossy\Program\Git\mingw64\bin;C:\Drossy\Program\Git\usr\local\bin;C:\Drossy\Program\Git\usr\bin;C:\Drossy\Program\Git\usr\bin;C:\Drossy\Program\Git\mingw64\bin;C:\Drossy\Program\Git\usr\bin;C:\Documents and Settings\Administrator\bin;C:\Drossy\Program\Ruby193\bin;C:\Drossy\Program\Python27\Scripts;C:\apache-tomcat-6.0.35\bin;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\Program Files\MySQL\MySQL Enterprise Backup 3.12;C:\Drossy\Program\Java\jdk1.7.0_72\bin;C:\Drossy\Program\Python27\python.exe;C:\Drossy\Program\nodejs;C:\Documents and Settings\Administrator\Application Data\npm;C:\Drossy\Program\Git\usr\bin\vendor_perl;C:\Drossy\Program\Git\usr\bin\core_perl
21515 verbose lifecycle node-sass@3.4.2~postinstall: CWD: C:\temp\node_modules\node-sass
21516 silly lifecycle node-sass@3.4.2~postinstall: Args: [ '/d /s /c', 'node scripts/build.js' ]
21517 silly lifecycle node-sass@3.4.2~postinstall: Returned: code: 128  signal: null
21518 info lifecycle node-sass@3.4.2~postinstall: Failed to exec postinstall script
21519 verbose unlock done using C:\Documents and Settings\Administrator\Application Data\npm-cache\_locks\staging-7d368c8e4ed0da7d.lock for C:\temp\node_modules\.staging
21520 silly rollbackFailedOptional Starting
21521 silly rollbackFailedOptional Finishing
21522 silly runTopLevelLifecycles Starting
21523 silly runTopLevelLifecycles Finishing
21524 silly install printInstalled
21525 warn temp@ No description
21526 warn temp@ No repository field.
21527 warn temp@ No license field.
21528 verbose stack Error: node-sass@3.4.2 postinstall: `node scripts/build.js`
21528 verbose stack Exit status 128
21528 verbose stack     at EventEmitter.<anonymous> (C:\Drossy\Program\nodejs\node_modules\npm\lib\utils\lifecycle.js:239:16)
21528 verbose stack     at emitTwo (events.js:100:13)
21528 verbose stack     at EventEmitter.emit (events.js:185:7)
21528 verbose stack     at ChildProcess.<anonymous> (C:\Drossy\Program\nodejs\node_modules\npm\lib\utils\spawn.js:24:14)
21528 verbose stack     at emitTwo (events.js:100:13)
21528 verbose stack     at ChildProcess.emit (events.js:185:7)
21528 verbose stack     at maybeClose (internal/child_process.js:850:16)
21528 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:215:5)
21529 verbose pkgid node-sass@3.4.2
21530 verbose cwd C:\temp
21531 error Windows_NT 5.2.3790
21532 error argv "C:\\Drossy\\Program\\nodejs\\node.exe" "C:\\Drossy\\Program\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install" "-ddd" "node-sass" "--sass-binary-site=https://npm.taobao.org/mirrors/node-sass/"
21533 error node v5.10.1
21534 error npm  v3.8.3
21535 error code ELIFECYCLE
21536 error node-sass@3.4.2 postinstall: `node scripts/build.js`
21536 error Exit status 128
21537 error Failed at the node-sass@3.4.2 postinstall script 'node scripts/build.js'.
21537 error Make sure you have the latest version of node.js and npm installed.
21537 error If you do, this is most likely a problem with the node-sass package,
21537 error not with npm itself.
21537 error Tell the author that this fails on your system:
21537 error     node scripts/build.js
21537 error You can get information on how to open an issue for this project with:
21537 error     npm bugs node-sass
21537 error Or if that isn't available, you can get their info via:
21537 error     npm owner ls node-sass
21537 error There is likely additional logging output above.
21538 verbose exit [ 1, true ]
```

I used another mirror for binary file because I cannot access the original one from local network, I've noticed in the README.
> Compiling versions 0.9.4 and above on Windows machines requires Visual Studio 2013 WD

Does it mean VS 2010 is not expected to work? As VS 2013 does not support windows xp, does it mean node-sass is not able to work on windows xp? Thanks.

**Comment by FisherYu-CN** [comment]
Changed to use node-sass 3.4.1 and it works fine. It seems 3.4.2 not work for windows server 2003/xp.

**Comment by hohwille** [comment]
Thanks @FisherYu-CN for the hint.
As I actually wanted to install reveal.js, I just changed its package.json to require node-sass 3.4.1 instead of 3.3.3 and now it is working.

Does this mean that both 3.3.3 and 3.4.2 are broken (on Win)?

**Comment by saper** [comment]
No, they are not. To compile node-sass from source few things need to be set up - the compiler and SDKs need to be installed, node-gyp should be able to install node development files etc.

Visual Studio 2013 is recommended because libsass is using advanced C++ features and this is what node itself is using. You can build the module on a newer Windows and use it on XP. If test us too complicated, just use a pre-built module.

**Comment by hohwille** [comment]
> No, they are not.

Then can you please explain why it works for 3.4.1 then without changing anything in my environment (node-gyp, compiler, SDKs)? This remains a mystery to me.

**Comment by xzyfer** [comment]
Node Sass v3.4.2 had a change that unintentionally made it incompatible with Windows XP. This has since been fixed in the v3.5.0-beta and the upcomming v3.5.0 stable release.

If you need Windows XP support please use v3.4.1 until v3.5.0 is released.

**Comment by saper** [comment]
Ah, I am sorry everyone: @xzyfer rightfully reminds me that the fix didn't make it into v3.4.2. Use `npm install node-sass@beta` to get the beta release.

**Comment by indralascu** [comment]
> Ah, I am sorry everyone: @xzyfer rightfully reminds me that the fix didn't make it into v3.4.2. Use `npm install node-sass@beta` to get the beta release.

Thank you so much! This really did it for me!
